main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -1264,14 +1264,24 @@ object Typer extends Phase[ResolvedAst.Root, TypedAst.Root] {
           */
         def inferSelectRule(sr0: ResolvedAst.SelectChannelRule): InferMonad[(List[Ast.TypeConstraint], Type, Type)] =
           sr0 match {
-            case ResolvedAst.SelectChannelRule(sym, chan, body) => for {
+            case ResolvedAst.SelectChannelRule.SelectGet(sym, chan, body) => for {
               (chanConstrs, chanType, _) <- visitExp(chan)
               (bodyConstrs, bodyType, _) <- visitExp(body)
               _ <- unifyTypeM(chanType, Type.mkChannel(sym.tvar, sym.loc), sym.loc)
               resultCon = chanConstrs ++ bodyConstrs
               resultTyp = bodyType
               resultEff = Type.Impure
             } yield (resultCon, resultTyp, resultEff)
+
+            case ResolvedAst.SelectChannelRule.SelectPut(chan, value, body) => for {
+              (chanConstrs, chanType, _) <- visitExp(chan)
+              (valueConstrs, valueType, _) <- visitExp(value)
+              (bodyConstrs, bodyType, _) <- visitExp(body)
+              // TODO: Should I unify something here?
```

Comment:
Yes,

You should merge all the "type class constraints" - i.e. chanConstrs, valueConstrs, and bodyConstrs.

It is okay to ignore the effects (as you do here with _), because SelectPut is *always* impure regardless of the purity of its constituents.